<launch>
  <arg name ="rviz_config"    default="$(find-pkg-share ike_slam)/config/ike_slam.rviz"/>

  <node_container pkg="rclcpp_components" exec="component_container" name="ike_slam" namespace="" args="">
    <composable_node pkg="ike_slam" plugin="ike_slam::IkeSlam" name="ike_slam" namespace="">
      <param name="loop_mcl_hz"                         value="10."/>
      <param name="particle_size"                       value="10"/>
      <param name="initial_pose_x"                      value="0.0"/>
      <param name="initial_pose_y"                      value="0.0"/>
      <param name="initial_pose_a"                      value="0.0"/>
      <param name="map_frame"                           value="map"/>
      <param name="odom_frame"                          value="odom"/>
      <param name="robot_frame"                         value="base_footprint"/>
      <param name="alpha_trans_trans"                   value="1."/>
      <param name="alpha_trans_rotate"                  value="0.2"/>
      <param name="alpha_rotate_trans"                  value="0.2"/>
      <param name="alpha_rotate_rotate"                 value="0.02"/>
      <param name="likelihood_dist"                     value="10."/>
      <param name="map_resolution"                     value="0.05"/>
      <param name="publish_particles_scan_match_point"  value="false"/>
      <extra_arg name="use_intra_process_comms"         value="false"/>
    </composable_node>
  </node_container>
  
  <node pkg="rviz2" exec="rviz2" name="rviz2" namespace="" args="-d $(var rviz_config)"/>
</launch>